src/test/java/uk/ac/ebi/ampt2d/accession/commons/core/BasicAccessioningServiceTest.java

```diff
@@ -0,0 +1,134 @@
+import uk.ac.ebi.ampt2d.accession.commons.hashing.SHA1HashingFunction;
+import uk.ac.ebi.ampt2d.accession.commons.persistence.BasicSpringDataRepositoryDatabaseService;
+import uk.ac.ebi.ampt2d.test.TestModel;
+import uk.ac.ebi.ampt2d.test.configuration.TestDatabaseServiceTestConfiguration;
+import uk.ac.ebi.ampt2d.test.persistence.TestEntity;
+import uk.ac.ebi.ampt2d.test.persistence.TestRepository;
+
+import java.util.Arrays;
+import java.util.Map;
+
+import static org.junit.Assert.assertEquals;
+
+@RunWith(SpringRunner.class)
+@DataJpaTest
+@ContextConfiguration(classes = {TestDatabaseServiceTestConfiguration.class})
+public class BasicAccessioningServiceTest {
+
+    @Autowired
+    private TestRepository repository;
+
+    @Autowired
+    private BasicSpringDataRepositoryDatabaseService<TestModel, TestEntity, String, String> databaseService;
+
+    @Test
+    public void testAccessionElements() throws AccessionCouldNotBeGeneratedException {
+        BasicAccessioningService<TestModel, String, String> accessioningService = getAccessioningService();
+
+        Map<String, TestModel> accessions = accessioningService.getOrCreateAccessions(Arrays.asList(
+                TestModel.of("service-test-1"),
+                TestModel.of("service-test-2"),
+                TestModel.of("service-test-3")
+        ));
+        assertEquals(3, accessions.size());
+    }
+
+    private BasicAccessioningService<TestModel, String, String> getAccessioningService() {
+        return new BasicAccessioningService<>(
+                SingleAccessionGenerator.ofHashAccessionGenerator(
+                        TestModel::getSomething,
+                        s -> "id-service-" + s
+                ),
+                databaseService,
+                TestModel::getSomething,
+                new SHA1HashingFunction()
+        );
+    }
+
+    @Test
+    public void testGetOrCreateFiltersRepeated() throws AccessionCouldNotBeGeneratedException {
+        BasicAccessioningService<TestModel, String, String> accessioningService = getAccessioningService();
+
+        Map<String, TestModel> accessions = accessioningService.getOrCreateAccessions(Arrays.asList(
+                TestModel.of("service-test-1"),
+                TestModel.of("service-test-2"),
+                TestModel.of("service-test-2"),
+                TestModel.of("service-test-3")
+        ));
+        assertEquals(3, accessions.size());
+    }
+
+    @Test
+    public void testGetAccessions() throws AccessionCouldNotBeGeneratedException {
+        BasicAccessioningService<TestModel, String, String> accessioningService = getAccessioningService();
+
+        Map<String, TestModel> accessions = accessioningService.getAccessions(Arrays.asList(
+                TestModel.of("service-test-1"),
+                TestModel.of("service-test-2"),
+                TestModel.of("service-test-3")
+        ));
+        assertEquals(0, accessions.size());
```

Comment:
why? the name of this method does not give any clues either.

Reply:
After reading the rest of the tests, I see that "get" just return the existing accessions, while "getOrCreate" creates them if they are not in the DB. Anyway this test would be clearer with a comment or a more detailed name.